frontend/src/stores/search.ts

```diff
@@ -266,8 +277,9 @@ export const useSearchStore = defineStore("search", {
       type: SupportedMediaType
       collectionParams: CollectionParams
     }) {
-      const path = `/${type}/${collectionToPath(collectionParams)}`
-      return this.$nuxt.localePath(path)
+      const path = `/${type}/collection`
+      const query = buildCollectionQuery(collectionParams, "frontend")
```

Comment:
```suggestion
      const { collection, ...params } = collectionParams
      const query = params
```

Just an example, this could be simpler and then `buildCollectionQuery` wouldn't need to support frontend and API modes, I think.

Reply:
Great idea! This will make the code much simpler, clearer and easier to understand :)